pyomo/contrib/mindtpy/MindtPy.py

```diff
@@ -121,6 +115,8 @@ def solve(self, model, **kwds):
         if config.solver_tee:
             config.mip_solver_tee = True
             config.nlp_solver_tee = True
+        if config.add_regularization in {'grad_lag', 'hess_lag'}:
+            config.equality_relaxation = True
```

Comment:
This is because the duals are only calculated when `equality_relaxation = True`. The old name of `equality_relaxation` is `use_dual`. Do you think we need a new option to obtain the duals of the NLPs?

Reply:
What do you need the duals for? To construct the Lagrangian? I think that that should come from the PyomoNLP model instead of the original NLP model.

Reply:
I just tested it. If we don't define the dual suffix in the original NLP model, the generated PyomoNLP model doesn't have dual values (all zero, using `get_duals()` function). So I think we still need to get the duals from the Pyomo original model.

Reply:
Perfectly reasonable. Do you think we should have separate options to calculate duals and for the equality relaxation? In a sense, the equality relaxation does much more than only compute the duals.